kura/org.eclipse.kura.driver.ble.sensortag.provider/src/main/java/org/eclipse/kura/internal/driver/ble/sensortag/TiSensorTag.java

```diff
@@ -0,0 +1,1503 @@
+                characteristics.addAll(entry.getValue().findCharacteristics());
+            } catch (KuraException e) {
+                logger.error("Failed to get characteristic", e);
+            }
+        }
+        return characteristics;
+    }
+
+    // ----------------------------------------------------------------------------------------------------------
+    //
+    // Temperature Sensor
+    //
+    // ----------------------------------------------------------------------------------------------------------
+    /*
+     * Enable temperature sensor
+     */
+    public void enableTermometer() {
+        // Write "01" to enable temperature sensor
+        byte[] value = { 0x01 };
+        BluetoothLeGattCharacteristic tempEnableChar;
+        try {
+            tempEnableChar = this.gattServices.get(TEMPERATURE)
+                    .findCharacteristic(TiSensorTagGatt.UUID_TEMP_SENSOR_ENABLE);
+            tempEnableChar.writeValue(value);
+        } catch (KuraException e) {
+            logger.error("Termometer enable failed", e);
+        }
+    }
+
+    /*
+     * Disable temperature sensor
+     */
+    public void disableTermometer() {
+        // Write "00" to disable temperature sensor
+        byte[] value = { 0x00 };
+        BluetoothLeGattCharacteristic tempEnableChar;
+        try {
+            tempEnableChar = this.gattServices.get(TEMPERATURE)
+                    .findCharacteristic(TiSensorTagGatt.UUID_TEMP_SENSOR_ENABLE);
+            tempEnableChar.writeValue(value);
+        } catch (KuraException e) {
+            logger.error("Termometer disable failed", e);
+        }
+    }
+
+    /*
+     * Read temperature sensor
+     */
+    public double[] readTemperature() {
+        double[] temperatures = new double[2];
+        BluetoothLeGattCharacteristic tempValueChar;
+        try {
+            tempValueChar = this.gattServices.get(TEMPERATURE)
+                    .findCharacteristic(TiSensorTagGatt.UUID_TEMP_SENSOR_VALUE);
+            temperatures = calculateTemperature(tempValueChar.readValue());
+        } catch (KuraException e) {
+            logger.error("Temperature read failed", e);
+        }
+        return temperatures;
+    }
+
+    /*
+     * Enable temperature notifications
+     */
+    public void enableTemperatureNotifications(Consumer<double[]> callback) {
+        Consumer<byte[]> callbackTemp = valueBytes -> callback.accept(calculateTemperature(valueBytes));
+
+        BluetoothLeGattCharacteristic tempValueChar;
+        try {
+            tempValueChar = this.gattServices.get(TEMPERATURE)
+                    .findCharacteristic(TiSensorTagGatt.UUID_TEMP_SENSOR_VALUE);
+            tempValueChar.enableValueNotifications(callbackTemp);
+        } catch (KuraException e) {
+            logger.error("Temperature notification enable failed", e);
+        }
+    }
+
+    /*
+     * Disable temperature notifications
+     */
+    public void disableTemperatureNotifications() {
```

Comment:
Similarly you can return boolean to signify the success of this method invocation.